runtime-modules/referendum/src/lib.rs

```diff
@@ -0,0 +1,740 @@
+// No default instance is provided.
+
+/////////////////// Configuration //////////////////////////////////////////////
+#![allow(clippy::type_complexity)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// used dependencies
+use codec::{Codec, Decode, Encode};
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage, error::BadOrigin, traits::Get, Parameter,
+    StorageValue,
+};
+use sp_arithmetic::traits::{BaseArithmetic, One};
+use sp_runtime::traits::{MaybeSerialize, Member};
+use std::marker::PhantomData;
+use system::ensure_signed;
+
+use std::collections::HashSet;
+
+// declared modules
+mod mock;
+mod tests;
+
+/////////////////// Data Structures ////////////////////////////////////////////
+
+#[derive(Encode, Decode, PartialEq, Eq, Debug)]
+pub enum ReferendumStage {
+    Void,
+    Voting,
+    Revealing,
+}
+
+impl Default for ReferendumStage {
+    fn default() -> ReferendumStage {
+        ReferendumStage::Void
+    }
+}
+
+#[derive(Encode, Decode, PartialEq, Eq, Debug, Default)]
+pub struct SealedVote<Hash, CurrencyBalance> {
+    commitment: Hash,
+    stake: CurrencyBalance,
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, Debug)]
+pub enum ReferendumResult<ReferendumOption, VotePower> {
+    Winners(Vec<(ReferendumOption, VotePower)>),
+    ExtraWinners(Vec<(ReferendumOption, VotePower)>),
+    NotEnoughWinners(Vec<(ReferendumOption, VotePower)>),
+    NoVotesRevealed,
+}
+
+impl<T, U> Default for ReferendumResult<T, U> {
+    fn default() -> ReferendumResult<T, U> {
+        ReferendumResult::NoVotesRevealed
+    }
+}
+
+/////////////////// Trait, Storage, Errors, and Events /////////////////////////
+
+pub trait Trait<I: Instance>: system::Trait {
+    /// The overarching event type.
+    type Event: From<Event<Self, I>> + Into<<Self as system::Trait>::Event>;
+
+    /// Maximum number of options in one referendum.
+    type MaxReferendumOptions: Get<u64>;
+    type ReferendumOption: Parameter
+        + Member
+        + BaseArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq
+        + From<u64>
+        + Into<u64>;
+
+    /// Currency balance used for stakes.
+    type CurrencyBalance: Parameter
+        + Member
+        + BaseArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq
+        + From<u64>
+        + Into<u64>;
+
+    type VotePower: Parameter
+        + Member
+        + BaseArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq
+        + From<u64>
+        + Into<u64>;
+
+    type VoteStageDuration: Get<Self::BlockNumber>;
+    type RevealStageDuration: Get<Self::BlockNumber>;
+
+    type MinimumStake: Get<Self::CurrencyBalance>;
+
+    fn is_super_user(account_id: &<Self as system::Trait>::AccountId) -> bool;
```

Comment:
There is no such concept in this module.

Reply:
The superuser role is useful when referendum module is used as standalone module. Will revise this mechanic again after I will introduce `on_finalized` logic to module.

Reply:
Superuser can run an extrinsic version of `start_referendum`. As stated before, this role makes sense when the referendum is used as a standalone module. Council module can omit this feature simply by implementing `is_super_user(...) {false}` in the runtime.